rfcs/51-animation-composition.md

```diff
@@ -0,0 +1,434 @@
+underlying clip is sampled at. For blend nodes, this value doesn't do anything.
+However, an optional setting will allow a blend node to propagate the time value
+to all of it's immediate children. If this option is set for it's chilren, it
+will continue to propagate downwards. This is disabled by default. Another option
+for advancing the animation is to advance the entire graph's time in one go. This
+will increment the time value on all nodes by a provided time delta multiplied by
+a clip node's.
+
+### Graph Edges
+Edges can be disconnected without losing metadata on it's input. This is
+functionally equivalent to setting a weight of 0, but the associated input node
+and it's descendants will not be evaluated, which can cut down CPU time spent
+evaluating that subgraph.
+
+### Example Graph
+
+```mermaid
+flowchart LR
+  root([Root])
+  mixerA([Mixer A])
+  mixerB([Mixer B])
+  clip01([Walk])
+  clip02([Run])
+  clip03([Jump Start])
+  clip04([Jump])
+  clip05([Jump End])
+  clip06([Punch])
+  mixerA-->root
+  mixerB-->root
+  clip06-->root
+  clip01-->mixerA
+  clip02-->mixerA
+  clip02-->mixerB
+  clip03-->mixerB
+  clip04-->mixerB
+  clip05-->mixerB
+```
+
+## Implementation strategy
+*Prototype implementation: https://github.com/HouraiTeahouse/bevy_prototype_animation*
+
+### Overview
+There are several systems that the animation system uses to drive animation from
+a graph:
+
+ 1. **Graph Evaluation**: This system evaluates the state of the graph to generate an
+    influences map that determines how the clips stored within the graph are going
+    to be blended. This can be done in parallel over every graph individually.
+    This is also when the internal clocks of the graph are ticked.
+ 2. **Binding**: This system traverses the entity hierarchy starting at the
+    graphs to find the corresponding entity for each bone animated by the graph.
+    This can be done in parallel over every graph individually.
+ 3. **Graph Sampling**: This system samples the graph for values from the active
+    clips, uses the generated influences map (from step 1) to blend them into
+    their final values, and applies it to the bound bones (from step 2).
+
+### Graph Storage
+Two main stores are needed internally for the graph: node storage and clip
+storage. Nodes are stored in a flat `Vec<Node>`, and a newtyped `NodeId` is used
+to keep a reference to a node within the graph. Due to this structure, nodes
+cannot be removed from the graph. As a worksaround, a node can be effectively
+removed by disconnecting all of it's incoming and outgoing edges. The root NodeId
+is always 0.
+
+Clip storage decomposes clips based on the properties each clip is animating.
+Instead of using `Handle<AnimationClip>`, each clip is instead internally
+assigned a auto-imcrementing `ClipId` much like `NodeId`. However, instead of
+storing a `Vec<AnimationClip`, a map of property paths to `Track` is stored.
```

Comment:
I would suggest adding a "this is explained in the 'Bindings' section". I didn't understand what "map of property paths to `Track`" meant until I read it (ie: the `GraphBindings` struct)